CareerCatalyst/Career-Catalyst
# 14 modularize guides

## Description
Modularized the guide cards & guide card container and added it to the landing page & the guides page:
1. Landing/Home page
![Screen Shot 2023-04-23 at 8 17 09 AM](https://user-images.githubusercontent.com/9134674/233841977-d55cbd9b-a064-4bed-86d1-c2be32d75cec.jpg)
2. Guides page
![Screen Shot 2023-04-23 at 8 19 42 AM](https://user-images.githubusercontent.com/9134674/233842089-8fc2ed37-0948-449c-9d6f-bc73cf9b4372.jpg)

## Related Issue
https://github.com/CareerCatalyst/Career-Catalyst/issues/14

## Acceptance Criteria

- [x] Created a separate guide card component
- [x] Update the `consants.js` file to include icon name and flag for coming soon to each guide topic
- [X] Add guide topic container to the `/guides` page  

## Type of Changes
Use a check for  ✓ the following type of changes: 

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|     | :sparkles: New feature     |
|  ✓    | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates
- `src/components/GuideCard.astro` Removed duplicate import
- `src/components/GuideCardContainer.astro` Created this guide card container so the container that holds the cards can be used on the landing page and also the Guides page.
- `src/components/GuidePageContainer.astro` Removed unused import
- `src/pages/guides/[post].astro` Deleted unused file
- `src/pages/guides/...` All the Guide files are updated to include the new import
- `src/pages/guides/index.astro` Added GuideCardContainer to reuse code
- `src/pages/index.astro` Cleaned up unused imports & added GuideCardContainer to reuse code
- `src/utils/constants.js` Added icon and published flag

## Testing Steps / QA Criteria
1. Navigate to the landing page
2. Click on a few cards
3. Confirm that you are navigated to the URL you expect & user doesn't receive 404 error.
4. Navigate to /guides url from the top nav
5. Repeat steps 2-3 on here
